improver/utilities/cube_manipulation.py

```diff
@@ -691,19 +691,18 @@ def enforce_coordinate_ordering(cube, coord_names, anchor_start=True):
         if coord in dim_coord_names:
             coords_to_reorder.append(coord)
 
-    # construct dictionary of original dimensions of the form, eg:
-    # {'time': 0, 'realization': 1, ...}
-    original_dims = {}
-    for coord in cube.coords(dim_coords=True):
-        original_dims[coord.name()] = cube.coord_dims(coord)[0]
+    original_coords = cube.coords(dim_coords=True)
+    coord_dims = cube.coord_dims
 
     # construct list of reordered dimensions assuming start anchor
-    new_dims = []
-    for coord in coords_to_reorder:
-        new_dims.append(original_dims[coord])
-    for coord in cube.coords(dim_coords=True):
-        if original_dims[coord.name()] not in new_dims:
-            new_dims.append(original_dims[coord.name()])
+    new_dims = [coord_dims(coord)[0] for coord in coords_to_reorder]
+    new_dims.extend(
+        [
+            coord_dims(coord)[0]
+            for coord in original_coords
+            if coord_dims(coord)[0] not in new_dims
+        ]
```

Comment:
you could use generator expression instead of list comprehension here

Reply:
Wouldn't it still need to be converted to a list before passing to ```cube.transpose```?

Reply:
> Wouldn't it still need to be converted to a list before passing to `cube.transpose`?

I meant `new_dims.extend(<generator>)` vs. `new_dims.extend(<list>)`. It avoids creating an intermediate list object (doesn't matter much here, because the list will be small, so it's just a style nitpick).